tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -1880,6 +1892,240 @@ tensorflow::Status ConvertReshape(OpConverterParams* params) {
   return tensorflow::Status::OK();
 }
 
+// Gets the bounds (start or end) from the weights of a StridedSlice op.
+tensorflow::Status GetStridedSliceBound(const std::vector<int>& input_dims,
+                                        const TRT_ShapedWeights& bound_weights,
+                                        string bound_name, string node_name,
+                                        std::vector<int>& output_bound) {
+  const int* weights_ptr =
+      static_cast<int*>(const_cast<void*>(bound_weights.GetValues()));
+  output_bound = 
+      std::vector<int>(weights_ptr, weights_ptr + bound_weights.count());
+  if (output_bound.size() != input_dims.size()) {
+    return tensorflow::errors::InvalidArgument(
+        "StridedSlice \"", bound_name, "\" specified ",
+        std::to_string(output_bound.size()), " dimensions, but input rank is ",
+        std::to_string(input_dims.size()), ", at ", node_name);
+  }
+  for (int i = 0; i < output_bound.size(); i++) {
+    // Make sure bound is valid.
+    if ((output_bound[i] < -input_dims[i]) ||
+        (output_bound[i] > input_dims[i])) {
+      return tensorflow::errors::InvalidArgument(
+          bound_name,
+          " for StridedSlice is invalid, must be in the range "
+          "[-rank(input), rank(input)], at ",
+          node_name);
+    }
+    // Convert negative values to their positive equivalent.
+    if (output_bound[i] < 0) {
+      output_bound[i] += input_dims[i];
+    }
+  }
+  return tensorflow::Status::OK();
+}
+
+tensorflow::Status ConvertStridedSlice(OpConverterParams* params) {
+  const auto& inputs = params->inputs;
+  const auto& node_def = params->node_def;
+  if (inputs.size() != 4) {
+    return tensorflow::errors::InvalidArgument(
+        "StridedSlice expects 4 inputs, at ", node_def.name());
+  }
+  if (!inputs.at(1).is_weights() || !inputs.at(2).is_weights() ||
+      !inputs.at(3).is_weights()) {
+    return tensorflow::errors::InvalidArgument(
+        "StridedSlice expects weights for begin, end, and strides, at ",
+        node_def.name());
+  }
+  if (!inputs.at(0).is_tensor()) {
+    return tensorflow::errors::Unimplemented(
+        "StridedSlice is only implemented for tensors, at ", node_def.name());
+  }
+  // Get input dims.
+  nvinfer1::Dims dims = inputs.at(0).GetTrtDims();
+  std::vector<int> input_dims(dims.d, dims.d + dims.nbDims);
+  if (inputs.at(0).is_tensor()) {
+    // Temporarily add batch dimension so that indexes line up properly.
+    input_dims.insert(input_dims.begin(), inputs.at(0).batch_size());
+  }
+  if (input_dims.size() > 4) {
```

Comment:
Is rank of smaller than 4 supported?

Reply:
Yes, we use reshape to add dimensions of size 1 to to make the tensor rank 4, and then reshape it back after the padding layer.